### ISSUE TITLE
Add support for wildcard locations for properties and YAML files

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-boot#19909]

**Comment by spencergibb** [comment]
I assume reading from the filesystem not using k8s api, correct?

**Comment by scottfrederick** [comment]
> I assume reading from the filesystem not using k8s api

Yes, consuming ConfigMaps mounted on the container filesystem using volume mounts.

**Comment by mbhave** [comment]
We discussed this a bit and it seems like adding a new property source isn't necessary. Here are some things we talked about:

- If the config map results in an `application.properties` file and you mount it in a `config/` directory next to the app, it’ll just work without us doing anything at all.
 
- If it's mounted at a different location, that location can be changed by setting the `SPRING_CONFIG_LOCATION` environment variable.

- If you have, let's say both redis configuration and some MySQL configuration, that could end up in `config/mysql/application.properties` and `config/redis/application.properties`. @dsyer had once suggested that the `ConfigFileApplicationListener` could support wildcards.

We've decided to go with the wildcard enhancement and documentation updates as part of this issue.
